### ISSUE TITLE
`TypeError` raised for `evalf` containing summations

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#21888]
```python3
from sympy import *


j = Symbol('j')
expr = 4*Sum(-1, (j, I, 5)) + 3*I*Sum(-1, (j, I, 5))
expr.evalf()
```

**Comment by oscarbenjamin** [comment]
What do you think `Sum(-1, (j, I, 5))` is supposed to represent? Probably it should give an immediate error.

**Comment by oscarbenjamin** [comment]
There are old open issues about disallowing non-integer limits for summations.

**Comment by anutosh491** [comment]
Yeah passing non-integer limits should raise an error which is not the case as of now . Something like `sympy.Sum(-1, (j, I, 5)).doit()` would return `-6 + I ` . Even something like `x = Symbol('x')` `sympy.Sum(1, (j, x, 5)).doit()` gives `6 - x` which maybe should raise a warning/exception or return `None` as we can't say anything about x as of now . But if `integer=True` is given then we should go for the appropriate answer.
Being relatively new to contributing to sympy ,I would be needing some guidance to approach this . If this issue is more towards the beginner/intermediate side I would try to take it on .

**Comment by faze-geek** [comment]
> There are old open issues about disallowing non-integer limits for summations.
@oscarbenjamin 
Is this a needs decision issue now , we can't disallow non-integer limits since many summations can be and should be evaluated by SymPy (eg -Sum(cos(n), (n, x, x + 1/2)).doit() ).  Should we raise the `TypeError` simply when one limit is complex and the other is real !?

**Comment by oscarbenjamin** [comment]
It's difficult to know what to do in terms of backwards compatibility but I think ideally the limits would be required to be integers (unless infinite).
